src/components/spreadsheet/utils/equipment-table-utils.ts

```diff
@@ -98,6 +107,55 @@ export const updateGeneratorCells = (params: CellEditingStoppedEvent) => {
     }
 };
 
+/*
+ * This function is used to format the data of the table to be able to display it in the table
+ * and resolve the issue of the calculated fields
+ */
+export const formatTwtDataForTable = (twt: any) => {
+    const formattedTwt = { ...twt };
+    if (formattedTwt?.ratioTapChanger) {
+        const regulationType = getComputedRegulationTypeId(formattedTwt);
+        const regulationSide =
+            regulationType === REGULATION_TYPES.LOCAL.id
+                ? getComputedTapSideId(formattedTwt)
+                : null;
+        const ratioRegulatingTerminal =
+            regulationType === REGULATION_TYPES.DISTANT.id
+                ? getTapChangerRegulationTerminalValue(
+                      formattedTwt.ratioTapChanger
+                  )
+                : null;
+        formattedTwt.ratioTapChanger = {
+            ...formattedTwt.ratioTapChanger,
+            regulationMode: getInitialTwtRatioRegulationModeId(twt),
+            regulationType: regulationType,
+            regulationSide: regulationSide,
+            ratioRegulatingTerminal: ratioRegulatingTerminal,
+        };
+    }
+    if (formattedTwt?.phaseTapChanger) {
+        const regulationType = getComputedPhaseRegulationTypeId(formattedTwt);
+        const regulationSide =
+            regulationType === REGULATION_TYPES.LOCAL.id
+                ? getPhaseTapRegulationSideId(formattedTwt)
+                : null;
+        const phaseRegulatingTerminal =
+            regulationType === REGULATION_TYPES.DISTANT.id
+                ? getTapChangerRegulationTerminalValue(
+                      formattedTwt.phaseTapChanger
+                  )
+                : null;
+        formattedTwt.phaseTapChanger = {
+            ...formattedTwt.phaseTapChanger,
+            regulationType: regulationType,
+            regulationSide: regulationSide,
+            phaseRegulatingTerminal: phaseRegulatingTerminal,
+        };
+    }
+
+    return formattedTwt;
```

Comment:
You can split this method into two: one for phaseTapChanger and one for the ratioTapChanger